src/getCommitMeaning.ts

```diff
@@ -8,22 +11,31 @@ const releaseCommitTester =
 	/^(?:chore(?:\(.*\))?:?)?\s*release|v?\d+\.\d+\.\d+/;
 
 export function getCommitMeaning(message: string) {
-	// Some types are always meaningful or ignored, regardless of potentially release-like messages
-	const { type } = conventionalCommitsParser.sync(message);
-	if (type) {
-		if (alwaysMeaningfulTypes.has(type)) {
+	let type = undefined;
+	try {
+		const messageAst = parser(message);
+		const commit = toConventionalChangelogFormat(messageAst);
+		type = commit.type;
+
+		// Some types are always meaningful or ignored, regardless of potentially release-like messages
+		if (
+			alwaysMeaningfulTypes.has(type) ||
+			commit.notes.some(({ title }) => title === "BREAKING CHANGE")
+		) {
 			return "meaningful";
 		}
 
 		if (alwaysIgnoredTypes.has(type)) {
 			return { type };
 		}
+	} catch {
+		/* empty */
```

Comment:
[Refactor] I'm not a big fan of `try`/`catch` as a control flow concept. It's not type-safe, is hard to mentally model, and can accidentally wrap around unintentionally throwing code too. Let's restrict what's being wrapped to just the `toConventionalChangelogFormat` call.